polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -194,120 +171,35 @@ impl Artifacts {
-		let _ = artifacts.insert_and_prune(cache_path).await.map_err(|err| {
-			gum::error!(
-				target: LOG_TARGET,
-				"could not initialize artifacts cache: {err}",
-			)
-		});
-		artifacts
-	}
-
-	async fn insert_and_prune(&mut self, cache_path: &Path) -> Result<(), String> {
-		async fn is_corrupted(path: &Path) -> bool {
-			let checksum = match tokio::fs::read(path).await {
-				Ok(bytes) => blake3::hash(&bytes),
-				Err(err) => {
-					// just remove the file if we cannot read it
-					gum::warn!(
-						target: LOG_TARGET,
-						?err,
-						"unable to read artifact {:?} when checking integrity, removing...",
-						path,
-					);
-					return true
-				},
-			};
-
-			if let Some(file_name) = path.file_name() {
-				if let Some(file_name) = file_name.to_str() {
-					return !file_name.ends_with(checksum.to_hex().as_str())
-				}
-			}
-			true
-		}
-
-		// Insert the entry into the artifacts table if it is valid.
-		// Otherwise, prune it.
-		async fn insert_or_prune(
-			artifacts: &mut Artifacts,
-			entry: &tokio::fs::DirEntry,
-			cache_path: &Path,
-		) -> Result<(), String> {
-			let file_type = entry.file_type().await;
-			let file_name = entry.file_name();
-
-			match file_type {
-				Ok(file_type) =>
-					if !file_type.is_file() {
-						return Ok(())
-					},
-				Err(err) => return Err(format!("unable to get file type for {file_name:?}: {err}")),
-			}
-
-			if let Some(file_name) = file_name.to_str() {
-				let id = ArtifactId::from_file_name(file_name);
-				let path = cache_path.join(file_name);
-
-				if id.is_none() || is_corrupted(&path).await {
-					let _ = tokio::fs::remove_file(&path).await;
-					return Err(format!("invalid artifact {path:?}, file deleted"))
-				}
-
-				let id = id.expect("checked is_none() above; qed");
-				gum::debug!(
-					target: LOG_TARGET,
-					"reusing existing {:?} for node version v{}",
-					&path,
-					NODE_VERSION,
-				);
-				artifacts.insert_prepared(id, path, SystemTime::now(), Default::default());
-
-				Ok(())
-			} else {
-				Err(format!("non-Unicode file name {file_name:?} found in {cache_path:?}"))
-			}
-		}
-
+	/// Create an empty table and the cache directory on-disk if it doesn't exist.
+	pub async fn new(cache_path: &Path) -> Self {
 		// Make sure that the cache path directory and all its parents are created.
-		if let Err(err) = tokio::fs::create_dir_all(cache_path).await {
-			if err.kind() != io::ErrorKind::AlreadyExists {
-				return Err(format!("failed to create dir {cache_path:?}: {err}"))
+		let _ = tokio::fs::create_dir_all(cache_path).await;
+
+		// Delete any leftover artifacts and worker dirs from previous runs. We don't delete the
+		// entire cache directory in case the user made a mistake and set it to e.g. their home
+		// directory. This is a best-effort to do clean-up, so ignore any errors.
+		if let Ok(paths) = fs::read_dir(cache_path) {
+			for path in paths.flat_map(|res| res.map(|e| e.path())) {
```

Comment:
You can reduce the level of indentation and simplify this a bit with something like this:

```rust
for entry in fs::read_dir(cache_path).into_iter().flatten().flatten() {
    let path = entry.path();
```